File: 1/main.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        int i;
        final HashMap<Integer, Integer> presented = new HashMap<Integer, Integer>();
        for (i = 0; i < nums.length; i++)
        {
            final int n = nums[i];
            final Integer j = presented.get(target - n);
            if (j != null)
            {
                final int[] result = new int[2];
                result[0] = i;
                result[1] = j;
                return result;
            }
            presented.put(n, i);
        }
        return null;
    }
}

File: 1/main.kt
class Solution {
    class UnreachableCodeException : RuntimeException()
    
    fun twoSum(nums: IntArray, target: Int): IntArray {
        val presented = HashMap<Int, Int>()
        nums.forEachIndexed { i, n -> 
            presented[target - n]?.also { j ->
                return@twoSum intArrayOf(i, j)
            }
            presented[n] = i
        }
        throw UnreachableCodeException()
    }
}
